src/hooks/useProfiles.ts

```diff
@@ -96,14 +73,37 @@ export const useDeleteProfile = (options?: UseMutationOptions<ServiceResponse<Co
   });
 };
 
+export const useProfileErrorHandler = () => {
+  const { t } = useTranslation('user');
+
+  return (
+    e: unknown,
+    setErrors: (
+      errors: Partial<
+        Omit<ProfilePayload, 'adult'> & {
+          adult: string;
+        } & GenericFormErrors
+      >,
+    ) => void,
+  ) => {
+    const formError = e as ProfileFormSubmitError;
+    if (formError.message.includes('409')) {
+      setErrors({ name: t('profile.validation.name.already_exists') });
+      return;
+    }
+    setErrors({ form: t('profile.form_error') });
+  };
+};
```

Comment:
You want to ensure that the `unknown` error is of type `ProfileFormSubmitError`. If you have control, even better would be to throw a custom error so that you can use `instanceof` checks instead.

There is already a ProfileFormValues type, why not use that in the `setErrors` argument?

```suggestion
export const isProfileFormSubmitError = (error: unknown): error is ProfileFormSubmitError => {
  return !!error && typeof error === 'object' && 'message' in error;
} 

export const useProfileErrorHandler = () => {
  const { t } = useTranslation('user');

  return (error: unknown, setErrors: (errors: Partial<ProfileFormValues & GenericFormErrors>) => void) => {
    if (isProfileFormSubmitError(error) && error.message.includes('409')) {
      setErrors({ name: t('profile.validation.name.already_exists') });
      return;
    }
    setErrors({ form: t('profile.form_error') });
  };
};
```